### ISSUE TITLE
munmap_chunk(): invalid pointer

### ISSUE THREAD

**Issue by fusion809** [AppImage/appimaged#31]
Hi,

On Tumbleweed appimaged (latest, from commit a3b100ba780eb67e950605dbf8d90c9b389bcd33) has started to crash, it used to run fine, but now I'm getting the error:

```bash
munmap_chunk(): invalid pointer
[1]    16855 abort (core dumped)  appimaged
```

. If you need any more information please do tell.

`strace -o strace.log appimaged` gives this strace.log: [strace.log.zip](https://github.com/AppImage/appimaged/files/2602782/strace.log.zip). `LD_DEBUG=libs appimaged &> debug.log` gives this debug.log: [debug.log.zip](https://github.com/AppImage/appimaged/files/2602786/debug.log.zip).

Thanks for your time,
Brenton

**Comment by fusion809** [comment]
I manually compiled the latest commit and it too gave this same error.

**Comment by fusion809** [comment]
I also get this issue on Arch Linux.

**Comment by probonopd** [comment]
Before a3b100b it was working on the same system?

**Comment by probonopd** [comment]
(Note to self, this is precisely why I don't like rolling release distributions - stuff that was tested yesterday can break suddenly today...)

**Comment by fusion809** [comment]
Yes to using the AppImage and a3b100b was when I noticed it, before then I had been using appimaged without a problem, although if it and a few earlier commits were pushed around the same time I wouldn't know which precise one caused it.

**Comment by TheAssassin** [comment]
@probonopd the commit only added some features on the CMake side, rather unlike it broke the AppImage.

@fusion809 you might want to try a few older builds https://artifacts.assassinate-you.net/artifactory/webapp/#/artifacts/browse/tree/General/appimaged. Perhaps you find a working one, then we can narrow down the commit range that causes the issues.

**Comment by fusion809** [comment]
Maybe it's a new library issue as I've gone back to build 33 from 27 August 2018 and still I'm getting this error.

**Comment by probonopd** [comment]
So let me rephrase my question @fusion809, has it ever worked? With which version of appimaged?

**Comment by fusion809** [comment]
@probonopd didn't you read where, in my [first comment](https://github.com/AppImage/appimaged/issues/31#issue-382964405), I said it did (use to work)? Not an attack, just thought you would have read my first comment. 

Commit https://github.com/AppImage/appimaged/commit/189b8000836c7425fa15548a3248dbf6b27ed30a worked, I am confident of that. I tried manually building it as an RPM and installing it, then running appimaged and I received the same error. So I'm guessing it is library related, or otherwise related to something newly introduced into roller distros. 

If you's try reproducing it, I should warn you, it doesn't appear immediately. Usually this error appears after it has built desktop config files for all AppImages in the searched locations (e.g. ~/Applications is where I store them).

**Comment by probonopd** [comment]
> didn't you read where, in my first comment, I said it did (use to work)? Not an attack, just thought you would have read my first comment.

Sorry, I must have overlooked this - working on AppImage only in my free time which happens to be very early in the mornings (when I am still half asleep) and/or late in the evenings (when I am already half asleep) ;-)

**Comment by fusion809** [comment]
Hmm, this is odd. On a separate PC, with an Arch Linux install I downloaded the appimaged AppImage, installed it, and it doesn't seem like I'm getting this issue.

**Comment by fusion809** [comment]
Seems fixed. The PC I originally had this issue on doesn't have it anymore.
